### PR TITLE
READY : Implementing & covering option `streamSizeLimit`.

### PR DESCRIPTION
**Pull request by JackieJoo**
1. Implemented option `streamSizeLimit`.
    streamSizeLimit : null, /* qqq for Yevhen : implement and cover the option. look option maxBuffer of spawn */
    1.1 Added asserts for the wrong usage of `streamSizeLimit`.
2. Wrote test routine `startOptionStreamSizeLimit`.
3. Wrote test routine `startOptionStreamSizeLimitThrowing`.
4. Fixed according to the review.

### CODE REVIEW

File: proto/wtools/abase/l4_process/l3/Execution.s [hunk 1]
```diff
@@ -137,6 +137,14 @@ function startCommon_head( routine, args )
   o.interpreterArgs = _.strSplitNonPreserving({ src : o.interpreterArgs });
   _.assert( o.interpreterArgs === null || _.arrayIs( o.interpreterArgs ) );
 
+  _.assert
+  (
+    ( _.numberIs( o.streamSizeLimit ) && o.streamSizeLimit >= 0 ) || o.streamSizeLimit === null, 'Option::streamSizeLimit must be a positive Number which is greater or equal to zero'
+  )
+
+  if( o.streamSizeLimit || o.streamSizeLimit === 0 )
+  _.assert( o.sync && ( o.mode === 'spawn' || o.mode === 'shell' ), 'Option::streamSizeLimit is allowed in mode::spawn and mode::shell with sync::1' )
```
Comment (Wandalen) [inline proto/wtools/abase/l4_process/l3/Execution.s:146]: is allowed -> is supported

File: proto/wtools/abase/l4_process/l3/Execution.s [hunk 2]
```diff
@@ -137,6 +137,14 @@ function startCommon_head( routine, args )
   o.interpreterArgs = _.strSplitNonPreserving({ src : o.interpreterArgs });
   _.assert( o.interpreterArgs === null || _.arrayIs( o.interpreterArgs ) );
 
+  _.assert
+  (
+    ( _.numberIs( o.streamSizeLimit ) && o.streamSizeLimit >= 0 ) || o.streamSizeLimit === null, 'Option::streamSizeLimit must be a positive Number which is greater or equal to zero'
```
Comment (Wandalen) [inline proto/wtools/abase/l4_process/l3/Execution.s:142]: too long line

File: proto/wtools/abase/l4_process/l3/Execution.s [hunk 3]
```diff
@@ -1242,6 +1251,8 @@ function startMinimal_body( o )
     }
     if( o.currentPath )
     o2.cwd = _.path.nativize( o.currentPath );
+
+    o2.maxBuffer = o.streamSizeLimit ? o.streamSizeLimit : 1024 * 1024;
```
Comment (Wandalen) [inline proto/wtools/abase/l4_process/l3/Execution.s:1255]: dont redefine maxBuffer if streamSizeLimit is null

File: proto/wtools/abase/l4_process/l3/Execution.s [hunk 4]
```diff
@@ -1225,6 +1233,7 @@ function startMinimal_body( o )
     if( o.timeOut && o.sync )
     o2.timeout = o.timeOut;
     o2.windowsHide = !!o.hiding;
+    o2.maxBuffer = o.streamSizeLimit ? o.streamSizeLimit : 1024 * 1024;
```
Comment (Wandalen) [inline proto/wtools/abase/l4_process/l3/Execution.s:1236]: dont redefine maxBuffer if streamSizeLimit is null

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 2 above)
Comment (Wandalen) [reply proto/wtools/abase/l4_process/l3/Execution.s:142]: greater than zero!

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 1 above)
Comment (Wandalen) [reply proto/wtools/abase/l4_process/l3/Execution.s:146]: assert should be covered! where is coverage?

**Review by Wandalen — CHANGES_REQUESTED** [review]

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 4 above)
Comment (JackieJoo) [reply proto/wtools/abase/l4_process/l3/Execution.s]: Done

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 3 above)
Comment (JackieJoo) [reply proto/wtools/abase/l4_process/l3/Execution.s]: Removed adding `maxBuffer` in routine `optionsForFork`. mode::fork doesn't support `maxBuffer`.

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 2 above)
Comment (JackieJoo) [reply proto/wtools/abase/l4_process/l3/Execution.s:142]: Refactored.

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 1 above)
Comment (JackieJoo) [reply proto/wtools/abase/l4_process/l3/Execution.s]: Renamed.
All asserts are already covered in the test routine `startOptionStreamSizeLimitThrowing`.

File: proto/wtools/abase/l4_process/l3/Execution.s [hunk 5]
```diff
@@ -137,6 +137,15 @@ function startCommon_head( routine, args )
...
   _.assert( o.interpreterArgs === null || _.arrayIs( o.interpreterArgs ) );
 
+  _.assert
+  (
+    ( _.numberIs( o.streamSizeLimit ) && o.streamSizeLimit > 0 ) || o.streamSizeLimit === null,
+    'Option::streamSizeLimit must be a positive Number which is greater than zero'
+  )
+
+  if( o.streamSizeLimit )
+  _.assert( o.sync && ( o.mode === 'spawn' || o.mode === 'shell' ), 'Option::streamSizeLimit is supported in mode::spawn and mode::shell with sync::1' )
```
Comment (Wandalen) [inline proto/wtools/abase/l4_process/l3/Execution.s:147]: too long line!

Comment (JackieJoo) [reply proto/wtools/abase/l4_process/l3/Execution.s]: Fixed

File: proto/wtools/abase/l4_process/l3/Execution.s (hunk 4 above)
Comment (Wandalen) [reply proto/wtools/abase/l4_process/l3/Execution.s]: No!

**Comment by Wandalen** [comment]
no loossy conditions

**Pull request merged**